webapp/src/js/enketo/widgets/db-object-widget.js

```diff
@@ -1,66 +1,57 @@
 'use strict';
 const _ = require('lodash/core');
-const Widget = require('enketo-core/src/js/Widget');
+const Widget = require('enketo-core/src/js/widget').default;
 const $ = require('jquery');
 const CONTACT_TYPE_CLASS_PREFIX = 'or-appearance-type-';
 
 require('enketo-core/src/js/plugins');
 
-const pluginName = 'dbobjectwidget';
+const mainSelector = '.or-appearance-db-object,.or-appearance-select-contact';
 
 /**
-   * Allows drop-down selectors for db objects.
-   *
-   * @constructor
-   * @param {Element} element [description]
-   * @param {(boolean|{touch: boolean, repeat: boolean})} options options
-   * @param {*=} e     event
-   */
-
-function Dbobjectwidget( element, options ) {
-  this.namespace = pluginName;
-  Widget.call( this, element, options );
-  this._init();
-}
-
-//copy the prototype functions from the Widget super class
-Dbobjectwidget.prototype = Object.create( Widget.prototype );
+ * Allows drop-down selectors for db objects.
+ *
+ * @extends Widget
+ */
+class Dbobjectwidget extends Widget {
+  static get selector() {
+    return `${mainSelector} input[type=text]`;
+  }
 
-//ensure the constructor is the new one
-Dbobjectwidget.prototype.constructor = Dbobjectwidget;
+  _init() {
+    construct(this.element);
+  }
 
-Dbobjectwidget.prototype._init = function() {
-  construct( this.element );
-};
+  list() {
+    return true;
+  }
+}
 
-function construct( element ) {
-  // timeout needed to let setting the value complete before rendering
-  setTimeout(function() {
-    const $question = $( element );
+function construct(element) {
+  const $question = $( element );
 
-    const Select2Search = window.CHTCore.Select2Search;
+  const Select2Search = window.CHTCore.Select2Search;
 
-    let $textInput = $question.find('input');
+  let $textInput = $question.find('input');
 
-    const value = $textInput.val();
-    const disabled = $textInput.prop('readonly');
-    $textInput.replaceWith($textInput[0].outerHTML.replace(/^<input /, '<select ').replace(/<\/input>/, '</select>'));
-    $textInput = $question.find('select');
-    const preSelectedOption = $('<option></option>')
-      .attr('value', value)
-      .text(value);
-    $textInput.append(preSelectedOption);
+  const value = $textInput.val();
+  const disabled = $textInput.prop('readonly');
+  $textInput.replaceWith($textInput[0].outerHTML.replace(/^<input /, '<select ').replace(/<\/input>/, '</select>'));
```

Comment:
It turns out that I don't think it was the timeout.  I had improperly updated the selector for the db-object-widget in such a way that in some cases (such as for the pregnancy form when the contact name of the user is loaded and shown) the widget would fail with an error.  (The call to setTimeout was just containing this error before and letting the form continue to load.)

I have reverted the changes to the selector and everything seem to load correctly now.